csharp/src/Ice/Properties.cs

```diff
@@ -0,0 +1,609 @@
+            string? line;
+            while ((line = input.ReadLine()) != null)
+            {
+                (string name, string value) = ParseLine(line);
+                if (name.Length > 0)
+                {
+                    into[name] = value;
+                }
+            }
+        }
+
+        /// <summary>Extracts the reserved Ice properties from command-line args.</summary>
+        /// <param name="into">The property dictionary into which the properties are added.</param>
+        /// <param name="args">The command-line args.</param>
+        public static void ParseIceArgs(this Dictionary<string, string> into, ref string[] args)
+        {
+            foreach (string name in PropertyNames.ClassPropertyNames)
+            {
+                into.ParseArgs(ref args, name);
+            }
+        }
+
+        /// <summary>Extracts properties from command-line args.</summary>
+        /// <param name="into">The property dictionary into which the parsed properties are added.</param>
+        /// <param name="args">The command-line args.</param>
+        /// <param name="prefix">Only arguments that start with --prefix are extracted.</param>
+        public static void ParseArgs(this Dictionary<string, string> into, ref string[] args, string prefix = "")
+        {
+            if (prefix.Length > 0 && !prefix.EndsWith(".", StringComparison.InvariantCulture))
+            {
+                prefix += '.';
+            }
+            prefix = "--" + prefix;
+
+            if ((prefix == "--" || prefix == "--Ice.") && Array.Find(args, arg => arg == "--Ice.Config") != null)
+            {
+                throw new ArgumentException("--Ice.Config requires a value", nameof(args));
+            }
+
+            var remaining = new List<string>();
+            var parsedArgs = new Dictionary<string, string>();
+            foreach (string arg in args)
+            {
+                if (arg.StartsWith(prefix, StringComparison.Ordinal))
+                {
+                    (string name, string value) = ParseLine((arg.Contains('=') ? arg : $"{arg}=1").Substring(2));
+                    if (name.Length > 0)
+                    {
+                        parsedArgs[name] = value;
+                        continue;
+                    }
+                }
+                remaining.Add(arg);
+            }
+
+            if ((prefix == "--" || prefix == "--Ice.") &&
+                    parsedArgs.TryGetValue("Ice.Config", out string? configFileList))
+            {
+                foreach (string file in configFileList.Split(","))
+                {
+                    into.LoadIceConfigFile(file);
+                }
+            }
+
+            foreach (KeyValuePair<string, string> p in parsedArgs)
+            {
+                into[p.Key] = p.Value;
+            }
+
+            args = remaining.ToArray();
+        }
+
+        internal static void CheckForUnknownProperties(this Communicator communicator, string prefix)
```

Comment:
Wouldn't make more sense to have this as a private static on the Reference?

Reply:
Yes, fixed.